midterm-2023-11-07/code/5130-stefan-vitanov/script.ts

```diff
@@ -0,0 +1,349 @@
+    movieContainer.appendChild(yearDiv);
+    movieContainer.appendChild(genreDiv);
+    movieContainer.appendChild(plotDiv);
+    movieContainer.appendChild(castDiv);
+    movieContainer.appendChild(oscarsDiv);
+ 
+    return movieContainer;
+  }
+ 
+ 
+  let currentSort: { field: keyof Movie | null, ascending: boolean } = {
+    field: null,
+    ascending: true
+  };
+ 
+  function sortMovies(movies: Movie[], sortField: keyof Movie, ascending: boolean): Movie[] {
+    return movies.sort((a, b) => {
+      let valA = a[sortField];
+      let valB = b[sortField];
+ 
+      if (sortField === 'genre') {
+ 
+        const genreLengthComparison = a.genre.length - b.genre.length;
+        if (genreLengthComparison !== 0) return ascending ? genreLengthComparison : -genreLengthComparison;
+ 
+ 
+        const sortedGenresA = [...a.genre].sort();
+        const sortedGenresB = [...b.genre].sort();
+        for (let i = 0; i < sortedGenresA.length; i++) {
+          const comparison = sortedGenresA[i].localeCompare(sortedGenresB[i]);
+          if (comparison !== 0) return ascending ? comparison : -comparison;
+        }
+ 
+        return 0; 
+      } else if (sortField === 'cast') {
+ 
+        const castLengthComparison = a.cast.length - b.cast.length;
+        if (castLengthComparison !== 0) return ascending ? castLengthComparison : -castLengthComparison;
+ 
+ 
+        for (let i = 0; i < a.cast.length; i++) {
+          const comparison = a.cast[i].actor.localeCompare(b.cast[i].actor);
+          if (comparison !== 0) return ascending ? comparison : -comparison;
+        }
+ 
+        return 0; 
+      } else if (typeof valA === 'string' && typeof valB === 'string') {
+ 
+        return ascending ? valA.localeCompare(valB) : valB.localeCompare(valA);
+      } else if (typeof valA === 'number' && typeof valB === 'number') {
+ 
+        return ascending ? valA - valB : valB - valA;
+      }
+ 
+ 
+      return 0;
+    });
+  }
+ 
+  async function populateGenreDropdown() {
+    const movies = await getMovies();
+    const genres = new Set(movies.flatMap(movie => movie.genre));
+    const genreSelect = document.getElementById('filter-genre');
+    genres.forEach(genre => {
+      const option = document.createElement('option');
+      option.value = genre;
+      option.textContent = genre;
+      if (genreSelect) {
+        genreSelect.appendChild(option);
+      }
+    });
+  }
+ 
+ 
+  async function populateOscarsDropdown() {
+    const movies = await getMovies();
+    const oscars = new Set(movies.flatMap(movie => Object.keys(movie.oscars)));
+    const oscarSelect = document.getElementById('filter-oscars');
+    oscars.forEach(oscar => {
+      const option = document.createElement('option');
+      option.value = oscar;
+      option.textContent = oscar.replace(/([A-Z])/g, ' $1').trim(); 
+      if (oscarSelect) {
+        oscarSelect.appendChild(option);
+      }
+    });
+  }
+ 
+ 
+  function applyFilters(movies: Movie[]) {
```

Comment:
The name of the method is wrong, it actually return the filtered data, without applying the filter.